src/main/java/Duke.java

```diff
@@ -1,10 +1,42 @@
+import duke.commands.Command;
+import duke.exceptions.DukeException;
+import duke.tasks.TaskList;
+import duke.util.Parser;
+import duke.util.Storage;
+import duke.util.Ui;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+
+    public Duke(String fileName) {
+        this.ui = new Ui();
+        this.storage = new Storage(fileName);
+        try {
+            taskList = new TaskList(storage.load());
+        } catch (DukeException e) {
+            System.out.println(e.toString());
+        }
     }
-}
+
+    public void run() throws DukeException {
+        ui.printWelcomeMsg();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                Command c = Parser.parseCommand(fullCommand);
+                c.execute(taskList, ui, storage);
+                isExit = c.isExit();
+            } catch (DukeException e) {
+                System.out.println(e.toString());
+            }
+        }
+
+    }
+
+    public static void main(String[] args) throws DukeException {
+        new Duke("duke.txt").run();
+    }
+}
```

Comment:
I like that you packaged your files rather neatly and your code is neat too! It's easy to follow along